### PR TITLE
chore: add support for development in a container and github codespaces

### PR DESCRIPTION
**Pull request by kranthie-sap**
# Value

To support developers who do not want to update `node`, `pnpm` and other software, this PR provides a `Dev Container` configuration file that contains recommended versions of the software that is required to build and test `open-ux-tools` in [GitHub Codespaces](https://github.com/features/codespaces) and in VS Code using the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers).

**Note**: Currently GitHub Codespaces offers 60 hours of free usage per month

# Description

* Add dev container config file
* Update `.gitignore` file to ignore `.pnpm-store` temporary folder generated in the container
* Update README file with information on remote development setup
* Update README file to fix linting errors

### CODE REVIEW

**Review by slin-sap — APPROVED** [review]

**Review by devinea — APPROVED** [review]
That's pretty useful. Thanks @kranthie-sap

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "image": "mcr.microsoft.com/devcontainers/typescript-node:1-16-bookworm",
+
+    // Features to add to the dev container. More info: https://containers.dev/features.
+    // "features": {},
+
+    // Use 'forwardPorts' to make a list of ports inside the container available locally.
+    // "forwardPorts": [],
+
+    // Use 'postCreateCommand' to run commands after the container is created.
+    "postCreateCommand": "curl -fsSL https://get.pnpm.io/install.sh | env PNPM_VERSION=8.6.0 SHELL=`which bash` bash -",
```
[the remark this answers is not in the document]
Comment (kranthie-sap) [reply .devcontainer/devcontainer.json:16]: @Klaus-Keller: The `pnpm` version in the dev container was `8.6.11`. This command was added to override the version to `8.6.0`.

[the remark this answers is not in the document]
Comment (kranthie-sap) [reply .devcontainer/devcontainer.json]: @Klaus-Keller: I changed this to use container features to use one with the required version of pnpm.

**Review by devinea — APPROVED** [review]
Thanks

**Review by tobiasqueck — APPROVED** [review]
* no changeset required
* documentation is clear
* did not test it locally

File: .devcontainer/devcontainer.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (kranthie-sap) [reply .devcontainer/devcontainer.json]: @Klaus-Keller: I'll merge the PR and can look at any feedback/improvements in another PR.

[the remark this answers is not in the document]
Comment (kranthie-sap) [reply .devcontainer/devcontainer.json]: I will mark this as resolved.

**Pull request merged**